server/src/main/java/io/crate/execution/engine/indexing/UpsertResults.java

```diff
@@ -182,6 +196,9 @@ private void updateErrorCount(String msg, List<Long> lineNumbers, Long increaseB
             if (errorEntry == null) {
                 errorEntry = new HashMap<>(1);
                 errors.put(msg, errorEntry);
+                if (errors.size() > MAX_ERRORS) {
+                    errors.remove(errors.keySet().iterator().next());
+                }
```

Comment:
Can we rather not add at all if `size == MAX_ERRORS`, which would be cheaper than remove?